### PR TITLE
Work on simple cache to cache identities in dictionary.

### PR DESCRIPTION
**Pull request by PremyslCerny**

### CODE REVIEW

**Comment by PremyslCerny** [comment]
@ateska Please see.

**Comment by PremyslCerny** [comment]
@ateska It is working, so I guess it can be merged, unless there are more comments.

File: asab/web/authn/oauth/middleware.py [hunk 1]
```diff
@@ -8,18 +9,23 @@
 #
 
 
-def oauthclient_middleware_factory(app, *args, methods, **kwargs):
+def oauthclient_middleware_factory(app, *args, methods, expiration_diff=60*60, **kwargs):
```
Comment (ateska) [inline asab/web/authn/oauth/middleware.py]: `expiration_diff` -> `identity_cache_longevity`

**Comment by PremyslCerny** [comment]
@ateska Renaming done.

File: asab/web/authn/oauth/middleware.py [hunk 2]
```diff
@@ -8,18 +9,23 @@
...
+	:identity_cache_longevity is an integer that specifies the number of seconds after which the cached identity expires
+
 	The expected format of Authorization header is:
 	Authorization: Bearer <OAUTH-SERVER-ID>-<ACCESS_TOKEN>
 
 	For more information about user info, visit:
 	https://connect2id.com/products/server/docs/api/userinfo
 	"""
 
+	cache = {}
```
Comment (ateska) [inline asab/web/authn/oauth/middleware.py]: `identity_cache = `

File: asab/web/authn/oauth/middleware.py [hunk 3]
```diff
@@ -37,12 +43,23 @@ def oauthclient_middleware_factory(app, *args, methods, **kwargs):
...
 		bearer = bearer_oauth[0]
+		oauth_server_id_access_token = bearer_oauth[1]
 
-		if "-" not in bearer_oauth[1]:
+		if "-" not in oauth_server_id_access_token:
 			L.warn("Authorization header's bearer '{}' is not in proper '<OAUTH-SERVER-ID>-<ACCESS_TOKEN>' format.".format(bearer_oauth[1]))
 			return await handler(request)
 
-		oauth_server_id, access_token = bearer_oauth[1].split('-', 1)
+		cached_identity_dict = cache.get(oauth_server_id_access_token)
```
Comment (ateska) [inline asab/web/authn/oauth/middleware.py]: `identity = cache.get(oauth_server_id_access_token)`

File: asab/web/authn/oauth/middleware.py [hunk 4]
```diff
@@ -37,12 +43,23 @@ def oauthclient_middleware_factory(app, *args, methods, **kwargs):
...
+		if cached_identity_dict is not None:
+			expiration = cached_identity_dict.get("Expiration", time.time())
```
Comment (ateska) [inline asab/web/authn/oauth/middleware.py]: Add a comment
# This is "cache hit" branch.

**Comment by PremyslCerny** [comment]
@ateska Renaming done.

**Pull request merged**